source/postprocess/boundary_velocity_residual_statistics.cc

```diff
@@ -0,0 +1,419 @@
+        // At the same time, calculate the rms velocity for each boundary
+        unsigned int index = 0;
+        for (std::set<types::boundary_id>::const_iterator
+             p = boundary_indicators.begin();
+             p != boundary_indicators.end(); ++p, ++index)
+          {
+            global_max_vel[*p] = global_max_values[index];
+            global_min_vel[*p] = global_min_values[index];
+            global_rms_vel[*p] = std::sqrt(global_velocity_square_integral_values[index] / global_boundary_area_values[index]);
+          }
+      }
+
+      // now add the computed max, min, and rms velocities to the statistics object
+      // and create a single string that can be output to the screen
+      std::ostringstream screen_text;
+      unsigned int index = 0;
+      for (std::map<types::boundary_id, double>::const_iterator
+           p = global_max_vel.begin(), a = global_min_vel.begin(), rms = global_rms_vel.begin();
+           p != global_max_vel.end() && a != global_min_vel.end() && rms != global_rms_vel.end();
+           ++p, ++a, ++rms, ++index)
+        {
+          if (this->convert_output_to_years() == true)
+            {
+              const std::string name_max = "Maximum residual velocity magnitude on boundary with indicator "
+                                           + Utilities::int_to_string(p->first)
+                                           + aspect::Utilities::parenthesize_if_nonempty(this->get_geometry_model()
+                                                                                         .translate_id_to_symbol_name (p->first))
+                                           + " (m/yr)";
+              statistics.add_value (name_max, p->second*year_in_seconds);
+              const std::string name_min = "Minimum residual velocity magnitude on boundary with indicator "
+                                           + Utilities::int_to_string(a->first)
+                                           + aspect::Utilities::parenthesize_if_nonempty(this->get_geometry_model()
+                                                                                         .translate_id_to_symbol_name (a->first))
+                                           + " (m/yr)";
+              statistics.add_value (name_min, a->second*year_in_seconds);
+              const std::string name_rms = "RMS residual velocity on boundary with indicator "
+                                           + Utilities::int_to_string(rms->first)
+                                           + aspect::Utilities::parenthesize_if_nonempty(this->get_geometry_model()
+                                                                                         .translate_id_to_symbol_name (rms->first))
+                                           + " (m/yr)";
+              statistics.add_value (name_rms, rms->second*year_in_seconds);
+              // also make sure that the other columns filled by the this object
```

Comment:
If you copied this from somewhere else, could you fix it there in a separate PR as well?

```suggestion
              // also make sure that the other columns filled by this object
```